File: app/src/main/java/intellij/class9/collections/exercise4/java/GuardaVolumes.java
package intellij.class9.collections.exercise4.java;

import java.util.HashMap;
import java.util.List;

public class GuardaVolumes {
    private int contador = 0;
    private HashMap<Integer, List<Peca>> lockers = new HashMap<>();

    public Integer guardarPecas(List<Peca> listaDePeca) {
        lockers.put(contador, listaDePeca);
        System.out.println("Identificador número: " + contador);
        contador++;
        return getContador();
    }

    public void mostrarPecas() {
        for (int chave : lockers.keySet()) {
            List<Peca> lista = lockers.get(chave);
            System.out.println("Guarda Volume " + chave + " tem:");
            for (Peca itemList : lista) {
                System.out.println("Peça: " + itemList.getMarca() + ", " + itemList.getModelo());
            }
        }
    }

    public void mostrarPecas(Integer numero) {
        List<Peca> lista = lockers.get(numero);
        System.out.println("No Guarda Volume " + numero + " tem: ");
        for (Peca itemList : lista) {
            System.out.println("Peça: " + itemList.getMarca() + ", " + itemList.getModelo());
        }
    }

    public void devolverPecas(Integer numero) {
        List<Peca> lista = lockers.remove(numero);
        System.out.println("Guarda Volume " + numero);
        for (Peca itemList : lista) {
            System.out.println("Peça: " + itemList.getMarca() + ", " + itemList.getModelo() + " devolvida");
        }
    }

    public int getContador() {
        return contador;
    }

    public void setContador(int contador) {
        this.contador = contador;
    }

    public HashMap<Integer, List<Peca>> getLockers() {
        return lockers;
    }

    public void setLockers(HashMap<Integer, List<Peca>> lockers) {
        this.lockers = lockers;
    }
}

File: app/src/main/java/intellij/class9/collections/exercise4/kotlin/GuardaVolumes.kt
package intellij.class9.collections.exercise4.kotlin

import java.util.HashMap

class GuardaVolumes {
    private var contador = 0
    private var lockers = HashMap<Int, List<Peca>>()

    fun guardarPecas(listaDePeca: List<Peca>): Int {
        lockers.put(contador, listaDePeca)
        println("Identificador número: $contador")
        contador++
        return contador
    }

    fun mostrarPecas() {
        for (chave in lockers.keys) {
            val lista = lockers[chave]
            println("Guarda Volume $chave tem:")
            lista?.let {
                for (itemList in lista) {
                    println("Peça: " + itemList.marca + ", " + itemList.modelo)
                }
            }
        }
    }

    fun mostrarPecas(numero: Int) {
        val lista = lockers[numero]
        println("No Guarda Volume $numero tem: ")
        lista?.let {
            for (itemList in it) {
                println("Peça: " + itemList.marca + ", " + itemList.modelo)
            }
        }
    }

    fun devolverPecas(numero: Int) {
        val lista = lockers.remove(numero)
        println("Guarda Volume $numero")
        lista?.let {
            for (itemList in it) {
                println("Peça: " + itemList.marca + ", " + itemList.modelo + " devolvida")
            }
        }

    }
}